libbeat/metric/system/process/process_linux.go

```diff
@@ -0,0 +1,22 @@
+package process
+
+import (
+	sigar "github.com/elastic/gosigar"
+)
+
+// InitSigarProcd keep initial value of gosigar Procd variable, that will be used for getting beats metrics
+var InitSigarProcd string
+var curSigarProcd string
+
+func init() {
+	InitSigarProcd = sigar.Procd
+}
+
+func (procStats *LocalStats) preGetOne() {
+	curSigarProcd = sigar.Procd
+	sigar.Procd = InitSigarProcd
```

Comment:
Solving it this way would probably require a locking mechanism in `gosigar`, as this value is global and there can be other goroutines using it at the same time.

Reply:
If this PR will be accepted in general, I think it can be done without big changes of gosigar.